File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int left = 0;
        int right = heights.length-1;
        int MaxArea = Integer.MIN_VALUE;

        while(left< right){
            int currArea = 0;
            int minimumHeight = Math.min(heights[left], heights[right]);

            currArea = minimumHeight * (right-left);

            if(heights[right]> heights[left]) left++;
            else right--;

            MaxArea = Math.max(currArea, MaxArea);
        }
        return MaxArea;
    }
}

File: Data Structures & Algorithms/sliding-window-maximum/submission-0.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:

        i,j=0,0
        ans = []
        
        while j<len(nums):
            
            if j-i+1<k:
               j+=1
            
            elif j-i+1==k:
                mx = ans.append(max(nums[i:j+1]))
                i+=1
                j+=1
        return ans
        